06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * InvertedIndex class
+ * @class
+ */
+class InvertedIndex {
+  /**
+   * class constructor
+   * @constructor
+   * @param {Array} filenames names of files to initialize the index with
+   * @param {Object} files file index(es) to be stored initially
+   */
+  constructor(filenames = [], files = {}) {
+    this.filenames = filenames;
+    this.files = files;
+  }
+
+  /**
+   * Validates file content
+   * @function
+   * @param {Array} fileContent content of uploaded file
+   * @return {Boolean} true if a valid file was uploaded, false otherwise
+   */
+  static validateFile(fileContent) {
+    if ((fileContent instanceof Array) && fileContent.length > 0) {
+      const BreakException = {};
```

Comment:
Hey man,

So awesome work here. I like what you've done!

### Nitpick:
So it'll help to add properties to the custom exception you're declaring. Such as `name`, `level`, `message` etc...

So you'd have:
```
const BreakError = {
  name: 'Validate File Error',
  message: 'File is invalid'
}
```
This way you can catch the error and check if it's the actual error you expected or that something else broke the execution and then handle that.

PS: it's `Errors` in JavaScript not the usual `Exceptions`